src/gridfs.rs

```diff
@@ -1,52 +1,71 @@
-#![allow(dead_code, unused_variables)]
 // TODO(RUST-1395) Remove these allows.
+#![allow(dead_code, unused_variables)]
 
+mod download;
 pub mod options;
 
 use core::task::{Context, Poll};
-use std::pin::Pin;
+use std::{
+    pin::Pin,
+    sync::{atomic::AtomicBool, Arc},
+};
+
+use serde::{Deserialize, Serialize};
+use tokio::io::ReadBuf;
 
 use crate::{
+    bson::{doc, oid::ObjectId, Binary, Bson, DateTime, Document},
     concern::{ReadConcern, WriteConcern},
     cursor::Cursor,
     error::Result,
-    selection_criteria::SelectionCriteria,
+    options::SelectionCriteria,
+    Collection,
     Database,
 };
-use bson::{oid::ObjectId, Bson, DateTime, Document};
+
 use options::*;
-use serde::{Deserialize, Serialize};
-use tokio::io::ReadBuf;
 
 pub const DEFAULT_BUCKET_NAME: &str = "fs";
 pub const DEFAULT_CHUNK_SIZE_BYTES: u32 = 255 * 1024;
 
 // Contained in a "chunks" collection for each user file
+#[derive(Debug, Deserialize, Serialize)]
 struct Chunk {
+    #[serde(rename = "_id")]
     id: ObjectId,
     files_id: Bson,
     n: u32,
     // default size is 255 KiB
-    data: Vec<u8>,
+    data: Binary,
 }
 
 /// A collection in which information about stored files is stored. There will be one files
```

Comment:
I know this is existing code, but I think this docstring could be a tad clearer. For instance, this struct models the documents in the files collection, rather than being the collection itself. Also, it could be better to say something like "metadata" rather than information, and also could be useful to mention the collection in which the files are actually stored (the "chunks" collection, right?).

Reply:
Follow up PR sounds good to me! Sorry, yeah I missed your previous comment.